Modules/ITS/include/ITS/ITSTrackTask.h

```diff
@@ -118,6 +125,149 @@ class ITSTrackTask : public TaskInterface
   double ptBins[141];     // pt bins
 
   o2::itsmft::TopologyDictionary* mDict;
+
+ private:
+  //analysis for its-only residual
+  o2::its::GeometryTGeo* mGeom;
+
+  double FitStepSize0 = 0.3;
+  double FitStepSize1 = 1.0e-5;
+  double FitStepSize2 = 1.0e-5;
+  double FitStepSize3 = 1.0e-5;
+
+  double FitTolerance = 1.0e-8;
+  double ITS_AbsBz = 0.5; //T
+
+  double inputG_C[3*NLayer];
+  double FitparXY[6];
+  double FitparDZ[2];
+  ROOT::Fit::Fitter fitterA;
+  ROOT::Fit::Fitter fitterB;
+  Int_t mAlignmentMonitor = 0;
+  Int_t mDefaultMomResPar = 0;
+  Int_t mResMonNclMin = 0;
+  float mResMonTrackMinPt = 0;
+
+  TH1D* hResidualRealTimePerTrackwFinerBin;
+  TH1D* hResidualCPUTimePerTrackwFinerBin;
+  std::array<std::unique_ptr<TH2D>, NLayer> hdxySensor{};//[NLayer];
+  std::array<std::unique_ptr<TH2D>, NLayer> hdzSensor{};//[NLayer];
```

Comment:
Do you that that we need TH2D in this case? The general issue which we are dealing with is the size of objects. Could you try configuration with TH2S or TH2I. In my understanding, you have very fine binning and we do not expect to have so many entries in each bin

Reply:
I changed this to TH2I, but I think TH2F can be more generally used in case we just take care of the size of objects.
Because the size of TH2I and TH2F is the same, but the maximum bin content for TH2F is more large.

If there is no problem using TH2F, I'd like to choose TH2F. Please leave your comments.